modules/service/rest-api-v2.js

```diff
@@ -526,6 +526,52 @@ class RestAPIServiceV2 {
         await this.dvController.handleDataReadRequest(data_set_id, reply_id, res);
     }
 
+    async _checkForReplicationHandlerStatus(req, res) {
+        const handler_object = await Models.handler_ids.findOne({
+            where: {
+                handler_id: req.params.handler_id,
+            },
+        });
+
+        if (handler_object == null) {
+            this.logger.info('Invalid request');
+            res.status(404);
+            res.send({
+                message: 'Unable to find data with given parameters!',
+            });
+            return;
+        }
+        const handlerData = JSON.parse(handler_object.data);
+
+        const offerData = {
+            holding_time_in_minutes: handlerData.holding_time_in_minutes,
+            token_amount_per_holder: handlerData.token_amount_per_holder,
+            status: handlerData.status,
```

Comment:
The status will already be given in the response, do not duplicate it unnecessarily.